source/navigator/AuthStack.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect, useState } from "react";
+import { createStackNavigator } from "@react-navigation/stack";
+import Onboarding from "../screens/onboarding";
+import LoginScreen from "../screens/LoginScreen";
+import StorageService, {
+  ONBOARDING_DISABLED,
+} from "../services/StorageService";
+
+const Stack = createStackNavigator();
+
+const AuthStack = (): JSX.Element | null => {
+  const [initialRouteName, setInitialRouteName] = useState<string | undefined>(
+    undefined
+  );
+
+  useEffect(() => {
+    const trySetInitialRouteName = async () => {
+      const isDisabled = await StorageService.getData(ONBOARDING_DISABLED);
+      const route = isDisabled ? "Login" : "Onboarding";
+
+      setInitialRouteName(route);
+    };
+
+    void trySetInitialRouteName();
+  }, []);
+
+  if (initialRouteName === undefined) return null;
```

Comment:
Do we allow one liners like this?
Prefer
`if(xxx) {
  return null;
}`

And you should avoid to return null. When we return null, we're creating work for ourselves and foisting problems upon our callers. All it takes is one missing null check to send an application spinning out of control.

It you're tempted to return null, consider throwing an exception instead.

Reply:
will fix